tools/torch/main.lua

```diff
@@ -370,15 +388,22 @@ end
 if opt.weights ~= '' then
     if paths.filep(opt.weights) then
         logmessage.display(0,'Loading weights from pretrained model - ' .. opt.weights)
-        local w, grad = model:getParameters()
-        w:copy(torch.load(opt.weights))
+        if (string.find(opt.weights, '_Weights')) then
+            local w, grad = model:getParameters()
+            w:copy(torch.load(opt.weights))
+        else
+            -- the full model was saved
+            assert(string.find(opt.weights, '_Model'))
```

Comment:
I have found problems with this approach when using external models that were not created in digits (we can thus finally use in digits now, since we dump the full model and not just the weights! Yay!
I would therefore just remove the assertion for _Model. Or maybe, if possible, load in the file and see if there's a model structure in there if you really want to be clean.
This approach as it is now requires someone to change the filename of his pretrained model.